go/vt/binlog/keyrange_filter.go

```diff
@@ -40,40 +30,23 @@ func KeyRangeFilterFunc(kit key.KeyspaceIdType, keyrange *pb.KeyRange, sendReply
 				log.Warningf("Not forwarding DDL: %s", statement.Sql)
 				continue
 			case proto.BL_DML:
-				keyspaceIndex := strings.LastIndex(statement.Sql, KEYSPACE_ID_COMMENT)
-				if keyspaceIndex == -1 {
+				keyspaceId, unfriendly, err := annotations.ParseSQLAnnotation(string(statement.Sql))
+				if err != nil {
```

Comment:
Yes the way to do this is to declare a specific constant error in the other file, document in the function that you can return that error in that case, and then when you call the method, check for all three cases with a switch on error (nil, the specific error, and default). I'd go both ways though, I don;t have a preference.